packages/engine-test-utils/src/__tests__/pods-core/GraphvizPod.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { tmpDir } from "@dendronhq/common-server";
+import { FileTestUtils, NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+import { GraphvizExportConfig, GraphvizExportPod } from "@dendronhq/pods-core";
+import fs from "fs-extra";
+import _ from "lodash";
+import path from "path";
+import { runEngineTestV5, WorkspaceOpts } from "../../engine";
+import { checkNotInString, checkString } from "../../utils";
+
+const setupBasic = async (opts: WorkspaceOpts) => {
+  const { wsRoot, vaults } = opts;
+  await NoteTestUtilsV4.createNote({
+    wsRoot,
+    vault: vaults[0],
+    fname: "parent",
+    body: [`[[foo.linked]]`].join("\n"),
+  });
+  await NoteTestUtilsV4.createNote({
+    wsRoot,
+    vault: vaults[0],
+    fname: "parent.linked",
+    body: [`[[foo]]`].join("\n"),
+  });
+  await NoteTestUtilsV4.createNote({
+    wsRoot,
+    vault: vaults[0],
+    fname: "unlinked",
+    body: [`[[parent]]`].join("\n"),
+  });
+};
+
+describe("graphviz export pod", () => {
+  let exportDest: string;
+
+  beforeAll(() => {
+    exportDest = tmpDir().name;
+  });
+
+  test("basic", async () => {
+    await runEngineTestV5(
+      async ({ engine, vaults, wsRoot }) => {
+        const pod = new GraphvizExportPod();
+        engine.config.useFMTitle = true;
+
+        await pod.execute({
+          engine,
+          vaults,
+          wsRoot,
+          config: {
+            dest: exportDest,
+            includeBody: true,
+            includeStubs: true,
+            includeHierarchy: true,
+            includeLinks: false,
+          },
+        });
+
+        // check that graphviz file is created
+        let [expectedFiles, actualFiles] = FileTestUtils.cmpFiles(exportDest, [
+          "graphviz.dot",
+        ]);
+        expect(expectedFiles).toEqual(actualFiles);
+
+        // check contents of graphviz file
+        const dotFile = fs.readFileSync(path.join(exportDest, "graphviz.dot"), {
+          encoding: "utf8",
+        });
+
+        // TODO: IDs of files switch between test runs, which causes the snapshot
```

Comment:
since we don't have a snapshot, it'll be good to have a comment at the top with how the basic graphs look like to make this easier to debug in the future

Reply:
Sounds good, I'll do that. Is there an easy way to maintain IDs between test runs so that we could use a snapshot here?

Reply:
so not yet - i'm thinking of making that an option to `runEngine`. currently, the non-root nodes have stable ids in the test but the root nodes do not